src/sql/run.py

```diff
@@ -391,6 +391,10 @@ def run(conn, sql, config, user_namespace):
                 result = FakeResultProxy(cur, headers)
             else:
                 txt = sqlalchemy.sql.text(statement)
+
+                if config.autocommit:
+                    conn.session.execution_options(isolation_level="AUTOCOMMIT")
+
```

Comment:
is this supported by all databases? my guess is that it's not and it'll break in some cases

Reply:
According to sqlalchemy [this is how we set the autocommit option](https://docs.sqlalchemy.org/en/20/orm/session_transaction.html#setting-transaction-isolation-levels-dbapi-autocommit). 

I think the question is if someone is setting `sql.autocommit = True` - is it the same as setting the autocommit option

Reply:
True, I think we need to dig deeper into this, possibly in sqlalchemy's docs

Reply:
for example, the error doesn't happen in SQLite, I can create a table without issues

Reply:
let's wait for https://github.com/ploomber/jupysql/issues/116 so we have a better idea on how this might affect other databases

also, I looked at the code. based on what I saw, I guess that the original author didn't use `execution_options(isolation_level="AUTOCOMMIT")` because it's [not compatible with all dbs](https://docs.sqlalchemy.org/en/20/core/connections.html#setting-isolation-level-or-dbapi-autocommit-for-a-connection), instead it's executing a commit command after running SQL.

Reply:
Test case added in https://github.com/ploomber/jupysql/pull/117 👍